bazaar/src/lib.rs

```diff
@@ -89,73 +88,113 @@ decl_module! {
-                // Get the index of the shop registry
-                Ok(shop_registry_index) => {
-                    match owned_shops.binary_search(&shop) {
-                        // If the search succeeds, delete the respective entries
-                        Ok(onwed_shops_index) => {
-                            // Remove the shop from the local registries
-                            owned_shops.remove(onwed_shops_index);
-                            shops.remove(shop_registry_index);
-                            // Update blockchain
-                            ShopsOwned::<T>::insert(cid, &sender, owned_shops);
-                            ShopRegistry::insert(cid, shops);
-                            ShopOwner::<T>::remove(cid, &shop);
-                            // Emit an event that the shop was removed
-                            Self::deposit_event(RawEvent::ShopRemoved(cid, sender, shop));
-                            Ok(())
-                        },
-                        // If the search fails, no such shop is owned
-                        Err(_) => Err(Error::<T>::NoSuchShop.into()),
-                    }
-                },
-                // If the search fails, no such shop is owned
-                Err(_) => Err(Error::<T>::NoSuchShop.into()),
-            }
+            Self::deposit_event(RawEvent::OfferingCreated(cid, sender, oid));
+
+            Ok(())
+        }
+
+        #[weight = 10_000]
+        pub fn update_offering(origin, cid: CommunityIdentifier, oid: OfferingIdentifier, url: PalletString) -> DispatchResult {
+            // Check that the extrinsic was signed and get the signer
+            let sender = ensure_signed(origin)?;
+            // Check that the supplied community is actually registered
+            ensure!(<encointer_communities::Module<T>>::community_identifiers().contains(&cid),
+                Error::<T>::InexistentCommunity);
+
+            let business_identifier = BusinessIdentifier{community_identifier: cid, business_account: sender.clone()};
+
+            ensure!(OfferingRegistry::<T>::contains_key(business_identifier.clone(), oid.clone()), Error::<T>::InexistentOffering);
+
+            OfferingRegistry::<T>::mutate(business_identifier, oid, |o| o.url = url);
+
+            Self::deposit_event(RawEvent::OfferingUpdated(cid, sender, oid));
+
+            Ok(())
+        }
+
+        #[weight = 10_000]
+        pub fn delete_offering(origin, cid: CommunityIdentifier, oid: OfferingIdentifier) -> DispatchResult {
+            // Check that the extrinsic was signed and get the signer
+            let sender = ensure_signed(origin)?;
+            // Check that the supplied community is actually registered
+            ensure!(<encointer_communities::Module<T>>::community_identifiers().contains(&cid),
+                Error::<T>::InexistentCommunity);
+
+            let business_identifier = BusinessIdentifier{community_identifier: cid, business_account: sender.clone()};
```

Comment:
Don't you think this would be a less verbose if you implemented the `new` constructor?

You could call it without the field annotations.